mesonbuild/dependencies/dev.py

```diff
@@ -505,6 +506,59 @@ def get_methods():
         return [DependencyMethods.SYSTEM]
 
 
+class JDKSystemDependency(ExternalDependency):
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
+        super().__init__('jdk', environment, kwargs)
+
+        m = self.env.machines[self.for_machine]
+
+        java_home = os.environ.get('JAVA_HOME')
+        if java_home is None:
+            mlog.debug('$JAVA_HOME is unset')
+            self.is_found = False
+            return
+
+        self.version = environment.detect_java_compiler(self.for_machine).version
+
+        platform_include_dir = JDKSystemDependency.__machine_info_to_platform_include_dir(m)
```

Comment:
```suggestion
        platform_include_dir = self.__machine_info_to_platform_include_dir(m)
```

self can be used to access all methods of a class in python, and it is idiomatic to do so.